### PR TITLE
DietPi-Software | Restic: is a fast and secure backup program

### PR DESCRIPTION
**Pull request by goldfix**
Added new backup software:
- Restic: is a fast and secure backup program

### CODE REVIEW

**Comment by MichaIng** [comment]
Thanks for your contribution.

Generally we do not implement software options which are nothing more than an `apt install package`, which is the case here. We do have a bunch of software options from old times, but actually are thinking about removing them. There is not much point in cloning the Debian package list and provide a simple wrapper for it. Users should know that DietPi is Debian, that hence the whole Debian package repository is available and how to install such a package.

File: dietpi/dietpi-software [hunk 1]
```diff
@@ -4568,6 +4574,18 @@ The install script will now exit. After applying one of the the above, rerun die
 			(( $RAM_PHYS < 924 && $(free -m | mawk '/^Swap:/{print $2;exit}') < 512 )) && /boot/dietpi/func/dietpi-set_swapfile 512
 		fi
 
+		if To_Install 207 # Restic
+		then
+			# APT Update
+			G_AGUP
+
+			# APT package
```
Comment (MichaIng) [inline dietpi/dietpi-software]:
```suggestion
			# APT package
```
This is not needed as it is done once before all software installs start.

**Comment by goldfix** [comment]
> 

Up tu you to decide. :)
The only advantage is the final force update (the restic version in debian repository  is very old).

Thanks

**Comment by MichaIng** [comment]
> The only advantage is the final force update (the restic version in debian repository is very old).

Then it would be probably better to install it from a different source in the first place. When mixing APT packages with internal auto-updates it is possible that an APT upgrade downgrades your software. Also of course package integrity checks fails etc.

How are these internal updates done? A check + in case update on every `restic` call? An additional update service, timer or cron job?

**Comment by MichaIng** [comment]
@StephanStS @Joulinar 
Let's have this discussion here instead of in docs: Adding this would basically contradict previous decisions to not add software which can be installed just via APT.

The only other thing that is done here, is updating the binary from official compiled binaries. As this breaks the integrity of the package, I would then suggest to download the official binary from GitHub directly instead of installing the package. The question is whether upstream binaries are so much better to justify the implementation as alternative to APT. Another example where we did this was Rclone.

File: dietpi/dietpi-software [hunk 2]
```diff
@@ -4594,6 +4600,15 @@ The install script will now exit. After applying one of the the above, rerun die
 			(( $RAM_PHYS < 924 && $(free -m | mawk '/^Swap:/{print $2;exit}') < 512 )) && /boot/dietpi/func/dietpi-set_swapfile 512
 		fi
 
+		if To_Install 207 # Restic
+		then
+			# APT package
+			G_AGI restic
+
+			# Force autoupdate
+			G_EXEC restic self-update
```
Comment (MichaIng) [inline dietpi/dietpi-software]: This requires a change in the `Download_Install` function, I hope to find time to implement it later:
```suggestion
			case $G_HW_ARCH in
				3) local arch='arm64';;
				10) local arch='amd64';;
				*) local arch='arm';;
			esac

			local fallback_url="https://github.com/restic/restic/releases/download/v0.15.2/restic_0.15.2_linux_$arch.bz2"
			Download_Install "$(curl -sSfL 'https://api.github.com/repos/restic/restic/releases/latest' | mawk -F\" "/^ * \"browser_download_url\": \".*\/restic_[^\"\/]*_linux_$arch\.bz2\"$/{print \$4}")" /usr/local/bin/restic
```

File: dietpi/dietpi-software [hunk 3]
```diff
@@ -14069,6 +14084,11 @@ _EOF_
 			G_EXEC rm -Rf /{root,home/*}/.config/rclone
 		fi
 
+		if To_Uninstall 207 # Restic
+		then
+			G_AGP restic
```
Comment (MichaIng) [inline dietpi/dietpi-software]:
```suggestion
			[[ -f '/usr/local/bin/restic' ]] G_EXEC rm /usr/local/bin/restic
```

File: dietpi/dietpi-software [hunk 4]
```diff
@@ -759,6 +759,12 @@ Available commands:
 
 		# Cloud & Backup
 		#--------------------------------------------------------------------------------
+		software_id=207
+		aSOFTWARE_NAME[$software_id]='Restic'
+		aSOFTWARE_DESC[$software_id]='Restic is a modern backup program that can back up your files'
```
Comment (MichaIng) [inline dietpi/dietpi-software]: Following closer the description on their GitHub + making clear that it is CLI: https://github.com/restic/restic
```suggestion
		aSOFTWARE_DESC[$software_id]='Fast, efficient and secure command-line backup program'
```

**Comment by MichaIng** [comment]
Install test on all platforms: https://github.com/MichaIng/DietPi/actions/runs/4901851720

**Comment by MichaIng** [comment]
Looks good, merging it now.

**Comment by lima1** [comment]
@MichaIng , I had a restic binary in /usr/local/bin (not sure it's relevant, but mention it in case it is). After installing it with dietpi-software, the binary was updated, but was missing the executable flag. After chmod +x, it worked.

**Comment by MichaIng** [comment]
Ayy, many thanks for reporting. Fixed for next release: https://github.com/MichaIng/DietPi/commit/6ba1d61
I'll probably push a life patch for this.

**Pull request merged**